ext/drwrap/drwrap.c

```diff
@@ -1931,8 +1931,21 @@ drwrap_in_callee(void *arg1, reg_t xsp _IF_NOT_X86(reg_t lr))
                 break; /* we do need a post-call hook */
             }
         }
-        if (intercept_post && wrapcxt.retaddr != NULL)
+        if (intercept_post && wrapcxt.retaddr != NULL) {
+            if (!TEST(DRWRAP_REPLACE_RETADDR, wrap->flags)) {
```

Comment:
Hmm, so this if() is b/c pt->wrap_level is examined inside drwrap_ensure_postcall() if the flag is set (must be why the wrap_level++ was moved in the first place).  It feels a little fragile though if future code inside drwrap_ensure_postcall() is added which looks at wrap_level.

Maybe instead pass pt to drwrap_mark_retaddr_for_instru() and have it decrement right before its call to dr_redirect_execution()?  That has its own fragileness if it's called from somewhere else in the future -- but IMHO it seems nicer (could put a function-level comment explaining that if it doesn't return it decrements wrap_level under the assumption that it's re-starting in-callee).